### PR TITLE
Cleanup legacy handling for TRY

### PR DESCRIPTION
**Pull request by wenleix**
TRY expression used to be handled in a special way similar to lambda
expressions. #8889 remove this special handling by making TRY expression
a syntactic sugar using lambda.

However we are still keeping the name `LambdaAndTryExpression`
in many places, and keeps the `tryMethodMap` although it is never used.

This commit cleanup these legacy handling.

### CODE REVIEW

**Comment by martint** [comment]
Needs a more detailed commit message.

**Comment by electrum** [comment]
The tense used in the description makes it a bit confusing. We can rephrase it to be clearer:
```
The TRY expression used to be handled in a special way similar to lambda
expressions. #8889 removed this special handling by making the TRY expression
a syntactic sugar using lambdas.

This commit cleans up the leftover name `LambdaAndTryExpression` and
removes the unused `tryMethodMap`.
